File: learn_py_with_js/error/try.py
try:
    with open('file-not-exist.txt', 'r') as f:
        content = f.read()
        lines = content.split('\n')

        for line in lines:
            if line.strip() != '':
                print(line)
        first_line = lines[0].strip()
        print(f'First line: {first_line}')

        numbers = [int(line) for line in lines]
        print(f'Sum of numbers: {numbers}')

except Exception as error:
    print(f'An error occurred: {str(error)}')

File: learn_py_with_js/error/try.js
const fs = require('fs')
try {
    const content = fs.readFileSync('file-not-exist.txt', 'utf8')
    console.log('content:', content)
    const lines = content.split('\n')
    console.log('lines:', lines)

    for (const line of lines) {
        if (line.trim() !== '') {
            console.log(line)
        }
    }

    const firstLine = lines[0].trim();
    console.log(`First line: ${firstLine}`);

    const numbers = lines.map(Number);
    console.log('numbers:', numbers)
    console.log(`Sum of numbers: ${numbers.reduce((a, b) => a + b, 0)}`);

} catch (error) {
    console.error(`An error occurred: ${error.message}`);
}